webui/react/src/utils/cluster.ts

```diff
@@ -1,20 +1,34 @@
-import { Agent, isDeviceType, ResourceState, ResourceType } from 'types';
+import { Agent, ResourceState, ResourceType } from 'types';
 
 export const getSlotContainerStates = (
   agents: Agent[],
   resourceType: ResourceType,
   resourcePoolName?: string,
 ): ResourceState[] => {
   const slotContainerStates = agents
-    .filter((agent) => (resourcePoolName ? agent.resourcePools?.includes(resourcePoolName) : true))
-    .map((agent) => {
-      return isDeviceType(resourceType)
-        ? agent.resources.filter((res) => res.type === resourceType)
-        : agent.resources;
+    .filter((agent) => {
+      if (resourcePoolName === undefined) {
+        return true;
+      }
+      for (const agentResourcePool of agent.resourcePools) {
+        if (!resourcePoolName.includes(agentResourcePool)) {
+          return false;
+        }
+      }
+      return true;
```

Comment:
can we keep this filter the same as it was before? the `!resourcePoolName.includes(agentResourcePool) -> false` thing is kind of hard to follow and not equivalent to the previous `agentResourcePools?.includes(resourcePoolName)`.